### PR TITLE
Fix/timestamp ms

### PR DESCRIPTION
**Pull request by sleepdefic1t**
## Proposed changes

This assumes fix #21 is included for #20 

Timestamps strings for V2 tests were missing milliseconds, resulting in failed tests.
```
cpp-client/test/api/two/two_blocks.cpp:109: Failure
Expected equality of these values:
  "2017-03-21T13:00:00Z"
  human
    Which is: "2017-03-21T13:00:00.000Z"
[  FAILED  ] api.test_two_block (188 ms)
```
```
cpp-client/test/api/two/two_transactions.cpp:84: Failure
Expected equality of these values:
  "2018-08-24T15:54:26Z"
  human
    Which is: "2018-08-24T15:54:26.000Z"
```
```
cpp-client2/test/api/two/two_votes.cpp:84: Failure
Expected equality of these values:
  "2018-08-24T15:54:27Z"
  human
    Which is: "2018-08-24T15:54:27.000Z"
```
```
[----------] Global test environment tear-down
[==========] 62 tests from 1 test case ran. (24074 ms total)
[  PASSED  ] 59 tests.
[  FAILED  ] 3 tests, listed below:
[  FAILED  ] api.test_two_block
[  FAILED  ] api.test_two_transaction
[  FAILED  ] api.test_two_vote

 3 FAILED TESTS
```

#

After adding ms, all tests pass.

> `test/api/two/two_blocks.cpp`
```diff
-    ASSERT_STREQ("2017-03-21T13:00:00Z", human);
+    ASSERT_STREQ("2017-03-21T13:00:00.000Z", human);
```

> `test/api/two/two_transactions.cpp`
```diff
-    ASSERT_STREQ("2018-08-24T15:54:26Z", human);
+    ASSERT_STREQ("2018-08-24T15:54:26.000Z", human);
```

> `test/api/two/two_votes.cpp`
```diff
-    ASSERT_STREQ("2018-08-24T15:54:27Z", human);
+    ASSERT_STREQ("2018-08-24T15:54:27.000Z", human);
```

```
[----------] Global test environment tear-down
[==========] 62 tests from 1 test case ran. (23344 ms total)
[  PASSED  ] 62 tests.
```

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [ ] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Checklist

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/guidebook/contribution-guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by sleepdefic1t** [comment]
Good catch, @ciband 

> https://dexplorer.ark.io:8443/api/v2/blocks/13114381566690093367
```json
{
  "data": {
    "id": "13114381566690093367",
    "version": 0,
    "height": 1,
    "previous": null,
    "forged": {
      "reward": 0,
      "fee": 0,
      "total": 0,
      "amount": 1.25e+16
    },
    "payload": {
      "hash": "2a44f340d76ffc3df204c5f38cd355b7496c9065a1ade2ef92071436bd72e867",
      "length": 11395
    },
    "generator": {
      "username": null,
      "address": "D6Z26L69gdk9qYmTv5uzk3uGepigtHY4ax",
      "publicKey": "03d3fdad9c5b25bf8880e6b519eb3611a5c0b31adebc8455f0e096175b28321aff"
    },
    "signature": "3044022035694a9b99a9236655c658eb07fc3b02ce5edcc24b76424a7287c54ed3822b0602203621e92defb360490610f763d85e94c2db2807a4bd7756cc8a6a585463ef7bae",
    "transactions": 52,
    "timestamp": {
      "epoch": 0,
      "unix": 1490101200,
      "human": "2017-03-21T13:00:00.000Z"
    }
  }
}
```

AFAIK, ISO8601 does support ms.
Docs don’t show ms, but it’s definitely returning with ms.

What might be a good question is if Unix time is calculated using ms precision in Core.
Arduino’s `time_t` does lack ms though.

@faustbrian any idea if anything recently got changed pertaining to Core and timestamps?

**Comment by sleepdefic1t** [comment]
Tied to [Core #1106](https://github.com/ArkEcosystem/core/pull/1106) ?

**Comment by ciband** [comment]
time_t is "typically" (read: every platform i've ever touched) is a seconds since unix epoch (1/1/1970).

https://en.cppreference.com/w/c/chrono/time_t

IoT nor desktop should be putting milliseconds in here.  If we are using ISO8601 for timestamps and we need ms precision, I would vote to track the milliseconds separately (e.g. a double called "fractional") and then construct a composite value as needed for calculations or output.

If seconds are good enough, then that makes things simpler.

File: src/http/iot/http.cpp [hunk 1]
```diff
@@ -84,7 +84,7 @@ class PlatformHTTP : public AbstractHTTP
             http.setTimeout(3000);
             http.begin(this->host_, this->port_, request);
 
-            http.addHeader("Content-Type", "application/x-www-form-urlencoded");
+            http.addHeader("Content-Type", "application/json");
```
Comment (faustbrian) [inline src/http/iot/http.cpp]: This is from https://github.com/ArkEcosystem/cpp-client/pull/21 and doesn't belong here.

**Comment by faustbrian** [comment]
The human datetime was changed to ISO recently because we previously were using the format method of the lib we use which could change at anytime while an ISO string will be guaranteed to follow the standard.

The docs still need to be updated.

**Comment by ciband** [comment]
I did some more digging.  It does look like ISO8601 will support fractional seconds.  It was a cryptic footnote (at least to me).  Sorry for the previous mis-information.  

If core will be using millisecond precision then the cpp libraries will have to adapt.  I assume that is the intention?

**Comment by faustbrian** [comment]
The client should be adapted to the new ISO format, same will be done for all other clients soon.

File: src/http/iot/http.cpp (hunk 1 above)
Comment (sleepdefic1t) [reply src/http/iot/http.cpp]: removed

https://github.com/ArkEcosystem/cpp-client/pull/22/commits/3b33849a86586994e5dca2bb32392871e99e9bc8

File: test/api/two/two_blocks.cpp [hunk 2]
```diff
@@ -106,7 +106,7 @@ TEST(api, test_two_block)
     ASSERT_EQ(1490101200, timestampUnix);
 
     const char* human = timestamp["human"];
-    ASSERT_STREQ("2017-03-21T13:00:00Z", human);
+    ASSERT_STREQ("2017-03-21T13:00:00.000Z", human);
```
[the remark this answers is not in the document]
Comment (sleepdefic1t) [reply test/api/two/two_blocks.cpp]: Looks like docs do need updated 👍

**Comment by sleepdefic1t** [comment]
code from #21 removed.
This PR should resolve its failing test, @faustbrian

**Pull request merged**